app/mailers/spree/user_mailer.rb

```diff
@@ -1,7 +1,7 @@
 module Spree
   class UserMailer < BaseMailer
     def reset_password_instructions(user, token, *args)
-      @edit_password_reset_url = spree.edit_spree_user_password_url(:reset_password_token => token)
+      @edit_password_reset_url = spree.edit_spree_user_password_url(:reset_password_token => token, :host => Spree::Config.site_url)
```

Comment:
We never overwrite the `ActionMailer::Base.default_url_options[:host]` and that is set to demo.spreecommerce.com by default. So we'd either have to make this change or make `ActionMailer::Base.default_url_options[:host] = Spree::Config.site_url` somewhere else (probably outside of spree_auth_devise?).